http/watcher.go

```diff
@@ -117,10 +127,25 @@ func (w *BTCAddressWatcher) performRegisterPegIn(txHash string) error {
 	if err != nil {
 		return err
 	}
-	_, err = w.rsk.RegisterPegIn(opt, q, signature, rawTx, pmt, big.NewInt(int64(bh)))
+	err = w.rsk.RegisterPegInWithoutTx(q, signature, rawTx, pmt, big.NewInt(bh))
+	if err != nil {
+		if strings.Contains(err.Error(), "Failed to validate BTC transaction") {
+			return nil // allow retrying in case the bridge didn't acknowledge all required confirmations have occurred
+		}
+	}
+	tx, err := w.rsk.RegisterPegIn(opt, q, signature, rawTx, pmt, big.NewInt(bh))
 	if err != nil {
 		return err
 	}
+	ctx, cancel := context.WithTimeout(context.Background(), connectors.EthTimeout)
+	defer cancel()
+	s, err := w.rsk.GetTxStatus(ctx, tx)
+	if err != nil {
+		return nil
+	}
```

Comment:
Same as in performCallForUser

Reply:
same as above, timeout increased so the tx gvies an actual result most likely.